### PR TITLE
physics : completed list of non-SI units accepted by SI system but missing in SymPy

### LINKED ISSUE #22819: Non-SI units mentioned in the SI missing/incomplete in SymPy
I was going through the `physics ` module when I came across PR #22741 (which changed the state of l unit , added L in  `physics.units`) .I realised that though L is not an Si unit of volume (m^3 is)  it is mentioned in Wikipedia as a Non SI unit accepted by SI( ml is) . I feel there is a lack of such units for other physical quantities .
Like °C-Celsius ,f-fahrenheit for Temperature . Like cal-calorie for Energy . Like t-tonne for Mass 
I feel such units could be included since SymPy is very broadly used and currently some such units Min( minute ), L ( litre ) are already present in` physics/units/__init__.py`.

### PR DESCRIPTION
**Pull request by faze-geek**
#### References to other Issues or PRs

Closes #22819 

#### Brief description of what is fixed or changed
I have added 3 more Non-SI units which are internationally accepted by SI because of their widespread use. Such units were earlier partially present in the code. I have abided by this link - https://en.wikipedia.org/wiki/Non-SI_units_mentioned_in_the_SI
Appropriate tests have been added for the same .
#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by faze-geek** [comment]
Ping @oscarbenjamin @oscargus for review .
The changes look good to me .Merged PR #22741 was the backbone for this PR so I believe it won't require many changes .

File: sympy/physics/units/definitions/unit_definitions.py
Comment (oscargus) [inline sympy/physics/units/definitions/unit_definitions.py:120]: I think it should be 10**-27 (and many more decimal digits for 1.66). https://en.wikipedia.org/wiki/Dalton_(unit) (and note that 10**-27 is in kg.

Comment (oscargus) [reply sympy/physics/units/definitions/unit_definitions.py:120]: Compare to https://github.com/sympy/sympy/blob/0e4bab73562d6f90fcdf5fa079731f26a455347c/sympy/physics/units/systems/si.py#L165-L166 which is 1 Da.

Comment (faze-geek) [reply sympy/physics/units/definitions/unit_definitions.py:120]: Thanks .Will make these changes quick !

Comment (smichr) [reply sympy/physics/units/definitions/unit_definitions.py]: Is there a way to refer the Da to the amu so we only have the constant in one place?

File: sympy/physics/units/definitions/unit_definitions.py
Comment (smichr) [inline sympy/physics/units/definitions/unit_definitions.py:120]: Does this work? `Da.set_globale_relative_scale_factor(amu.scale_factor/kilo, kg)`

Comment (faze-geek) [reply sympy/physics/units/definitions/unit_definitions.py:120]: Yes makes, I think I should go ahead with this change .

File: sympy/physics/units/tests/test_quantities.py
Comment (smichr) [inline sympy/physics/units/tests/test_quantities.py:525]: How does this pass?
```python
>>> Da = Quantity('dalton',abbrev="Da")
>>> Da.scale_factor
1
>>> Da.set_global_relative_scale_factor(amu/kilo, kg)
>>> Da.scale_factor
atomic_mass_constant
>>> amu.scale_factor
1.66053906660000e-24
>>> amu == 1.66053906660000e-24
False
>>> Da.scale_factor == 1.66053906660000e-24
False
```

Comment (smichr) [reply sympy/physics/units/tests/test_quantities.py:525]: oh -- global scale vs quantity scale.

Comment (faze-geek) [reply sympy/physics/units/tests/test_quantities.py:525]: > How does this pass?

Surprised that this didn't give an assertion error in this first place .How should this be dealt with ?

Comment (smichr) [reply sympy/physics/units/tests/test_quantities.py:525]: But a quick look at other uses of global scale setting show args to be `(literal number, unit)`; it seems like it would be better to make the relative scale factor be `amu.scale_factor/kilo` and then the quantity scale setting would be unecessary? And it might be a good safeguard to do a check in global scale that the args are as expected: literal number, unit. So after `scale_factor = sympify(scale_factor)` we should verify that `scale_factor.is_number`.

Comment (smichr) [reply sympy/physics/units/tests/test_quantities.py:525]: This needs to be checked. Is our amu the same as the dalton or are we distinguishing between the unified amu and the old amu? Is it right to define a new quantity -- or is Da just an alias for (u)amu?

Comment (faze-geek) [reply sympy/physics/units/tests/test_quantities.py:525]: > This needs to be checked. Is our amu the same as the dalton or are we distinguishing between the unified amu and the old amu? Is it right to define a new quantity -- or is Da just an alias for (u)amu?

Yes ,this is why I had seperately defined dalton and not linked it with atomic_mass_constant earlier .Dalton was discovered much later than old amu and is referred as u(amu) .I believe old amu is much more of a universal constant than a unit.
https://en.m.wikipedia.org/wiki/Dalton_(unit)
https://chemistry.stackexchange.com/questions/62579/when-was-the-amu-renamed-the-dalton

Comment (faze-geek) [reply sympy/physics/units/tests/test_quantities.py:525]: @smichr I would request for some assistance on how to deal with this .

Comment (smichr) [reply sympy/physics/units/tests/test_quantities.py:525]: Our current definition for amu is using the same constant as you are proposing for Da so it seems to me that simply creating another alias for amu is in order: `Da = amu = atomic_mass_unit`. Maybe the old value could be added if necessary? `c12_amu`?

Comment (faze-geek) [reply sympy/physics/units/tests/test_quantities.py:525]: Yes this seems a fine fix, also avoiding unnecessary confusion . `Da = amu = atomic_mass_unit`.


PS. I'm currently dealing with a fever ,I'll work on this as soon as I get better !

Comment (faze-geek) [reply sympy/physics/units/tests/test_quantities.py:525]: Ping @smichr 
I have made the changes .Now Da is treated as an alias for amu . Thanks for the help and patience .

**Comment by faze-geek** [comment]
@oscargus @smichr  I beleive that all the required changes have been made .If you guys are satisfied can this be merged ?

**Comment by faze-geek** [comment]
Ping @oscarbenjamin I believe that all the required changes have been made .Could you add labels ,review it and merge if no there are no potential changes to be made !

**Comment by faze-geek** [comment]
Ping @smichr

**Comment by faze-geek** [comment]
Ping @moorepants 
This pr is seems fine from my side ,all corrections/suggestions have been handled . As a maintainer of physics module in SymPy , I would like to get your views .

**Comment by moorepants** [comment]
@smichr and @oscargus, you all reviewed this, looks fine on a brief scan to me, but maybe one of you can make the check and merge?

@faze-geek I am not a maintainer of the physics package, but only some sub-packages of physics. I've never committed any code to this one.

**Comment by faze-geek** [comment]
> @faze-geek I am not a maintainer of the physics package, but only some sub-packages of physics. I've never committed any code to this one.

My apologies ,I was unaware of that . Thanks !

**Comment by faze-geek** [comment]
Ping @sylee957 @smichr . LTGM !

**Review by dyc3 — APPROVED** [review]

**Comment by faze-geek** [comment]
@oscargus I feel the work in this pr is suitable and also approved . Would appreciate if you could look into it and merge If there are no potential changes needed .

**Comment by faze-geek** [comment]
cc @oscarbenjamin @akshanshbhatt 
Since this pr is also concerning the physics module and is looked on/approved by prior reviewers , I would like to know if this is good to go in ?

**Comment by faze-geek** [comment]
@oscargus Does it look good to you ? Seems fine from my side and @dyc3 has also approved ! Could go in if there are no potential changes .

**Comment by oscargus** [comment]
> Ping @sylee957 @smichr . LTGM !

Please avoid overuse of pinging. And claiming LGTM on your own work. Patience is required since we all have other things to do in our lives. (And overusing a small batch of reviewers will not help.)

(And, yes, I can also be a bit eager to get things in, but over time it has improved.)

File: sympy/physics/units/definitions/unit_definitions.py
Comment (oscargus) [inline sympy/physics/units/definitions/unit_definitions.py:123]: Weight are a bit confusing since the SI unit is kg and not gram, so not really clear if tonne should be Mg or kkg... Should there even be one? Compare with hectare which doesn't have one. (hundred are, one are = 100 m^2).

File: sympy/physics/units/systems/length_weight_time.py
Comment (oscargus) [inline sympy/physics/units/systems/length_weight_time.py:97]: Remove extra spaces.

**Comment by faze-geek** [comment]
> Please avoid overuse of pinging. And claiming LGTM on your own work. Patience is required since we all have other things to do in our lives. (And overusing a small batch of reviewers will not help.)
> 
> (And, yes, I can also be a bit eager to get things in, but over time it has improved.)

Sincere apologies from my side , will keep this in mind for future work .As far as pinging is considered I just see people who have helped/reviewed previous PR's with the same label ,which after your comment I assume is incorrect and will strictly refrain from now !

File: sympy/physics/units/definitions/unit_definitions.py
Comment (faze-geek) [reply sympy/physics/units/definitions/unit_definitions.py:123]: > Weight are a bit confusing since the SI unit is kg and not gram, so not really clear if tonne should be Mg or kkg... Should there even be one? Compare with hectare which doesn't have one. (hundred are, one are = 100 m^2).

Firstly I think you are talking about mass because kg and gram are it's units. You can checkout `unit_definitions.py` , where `set_global_relative_scale_factor ` has been without much attention to whether it's with the SI unit of the quantity or not ,like yard is set with respect to feet . If you are concerned with unit conversions then we always have` convert_to ` function and coincidently I have converted tonne to Mg in the test added .So I don't think we should have any issues here.
Secondly when I was making the changes in this file I saw that only units representing fundamental quantities had were subject to `set_global_relative_scale_factor` which makes sense . Hectare is a unit of area hence doesn't have one similar to why Litre for volume doesn't have one similar to why Newton for force doesn't have one .Even here I believe the current diff is appropriate.

**Comment by faze-geek** [comment]
Reopening to run tests again .

File: sympy/physics/units/definitions/unit_definitions.py
Comment (faze-geek) [reply sympy/physics/units/definitions/unit_definitions.py:123]: CC.@dyc3 This is the only unresolved issue in this PR . Since you are a contributor to `physics.units` I would like to have your thoughts on relevant comments !

Comment (dyc3) [reply sympy/physics/units/definitions/unit_definitions.py:123]: From what I can see, I think it would make more sense to have it be:

```suggestion
tonne.set_global_relative_scale_factor(mega, gram)
```

(nit) We should probably alias `tonne` to `metric_ton` as well.
(nit) I think `t` for `tonne` is confusing. I've never seen tonne referred to as t out in the wild.
(nit) I think you should prefer `kilogram` over `kg`.

Comment (faze-geek) [reply sympy/physics/units/definitions/unit_definitions.py:123]: 1. Adding `metric_ton` makes sense ,I will do that soon.
2. I looked into online literature for `t` .It is a valid symbol for `tonne `,checkout [wikipedia](https://en.wikipedia.org/wiki/Non-SI_units_mentioned_in_the_SI) .  So maybe I can make that change here but not remove all t instances from the diff.
3. Not sure where you would want me to prefer `kilogram` over `kg`. Since you have suggested to change `kg` into `gram` above.

Comment (dyc3) [reply sympy/physics/units/definitions/unit_definitions.py:123]: I meant prefer the longer name for a unit instead of the shorter abbreviated name. (so prefer `gram` over `g`) In general, I think it makes it easier to read. The only difference is cosmetic, hence the (nit).

Comment (faze-geek) [reply sympy/physics/units/definitions/unit_definitions.py:123]: Thanks .I get your point .

Comment (faze-geek) [reply sympy/physics/units/definitions/unit_definitions.py]: I have made a commit now ,I hope that the changes are appropriate !

Comment (faze-geek) [reply sympy/physics/units/definitions/unit_definitions.py]: I have made the changes and all checks have passed .Unless there are some particular doubts related to this I intend to resolve this issue soon . Thanks for the reviews !

**Review by dyc3 — APPROVED** [review]
Looks good to me!

**Comment by faze-geek** [comment]
Ping @Upabjojr 
I believe this pr is near to merge .Although you have not reviewed it earlier, I would be glad if you could check it out .Thanks !

**Comment by Upabjojr** [comment]
@faze-geek thanks for your contribution!

**Pull request merged**
